lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java

```diff
@@ -0,0 +1,45 @@
+package org.enso.syntax2;
+
+import org.enso.syntax2.serialization.Message;
+import java.io.File;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+public final class Parser implements AutoCloseable {
```

Comment:
Now, when the `Parser` is the main API, it is time to hide stuff that need not be visible from the generated classes:
- please hide `protected final` fields from `Token` & co. make them package private - e.g. without `protected`
- nobody is supposed to use `deserialize` directly - only the `Parser` - make those methods package private - e.g. remove `public`
- move `Message` into the same package as `Parser` and make it package private as well.
- probably move all the classes into the same package - one package is enough

The generated Javadoc would then be nicer.

Reply:
I'll make these changes in a separate PR, as it will require build-script coordination to move `GeneratedFormatTests` into the `org.enso.syntax2` package so that it can test package-private interfaces.

Reply:
See: #3605